R/recode-values.R

```diff
@@ -0,0 +1,506 @@
+#'
+#' @examples
+#' x <- c("NC", "NYC", "CA", NA, "NYC", "Unknown")
+#'
+#' # `recode_values()` is useful for fully recoding from one set of values to
+#' # another, creating an entirely new vector in the process. Note that any
+#' # unmatched values result in `NA`, or a `default` value.
+#' x |>
+#'   recode_values(
+#'     "NC" ~ "North Carolina",
+#'     "NYC" ~ "New York",
+#'     "CA" ~ "California"
+#'   )
+#'
+#' x |>
+#'   recode_values(
+#'     "NC" ~ "North Carolina",
+#'     "NYC" ~ "New York",
+#'     "CA" ~ "California",
+#'     default = "<not recorded>"
+#'   )
+#'
+#' # `replace_values()` is useful for updating an existing vector, tweaking a
+#' # few values along the way
+#' x |> replace_values("NYC" ~ "NY")
+#'
+#' # `replace_values()` is particularly nice for replacing `NA`s with values...
+#' x |> replace_values(NA ~ "Unknown (NA)")
+#' # ...or values with `NA`s
+#' x |> replace_values("Unknown" ~ NA)
+#'
+#' # Multiple values can be grouped within a single left-hand side to normalize
+#' # all problematic values at once
+#' x |> replace_values(c(NA, "Unknown") ~ "<not recorded>")
+#'
+#' # ---------------------------------------------------------------------------
+#' # Lookup tables
+#'
+#' # `recode_values()` works with more than just character vectors. Imagine you
+#' # have this series of Likert Scale scores, which is a scoring system that is
+#' # ordered from 1-5.
+#' data <- tibble(
+#'   score = c(1, 2, 3, 4, 5, 2, 3, 1, 4)
+#' )
+#'
+#' # To recode each `score` to its corresponding Likert Score label, you may
+#' # initially be inclined to reach for `case_when()`
+#' data |>
+#'   mutate(
+#'     score = case_when(
+#'       score == 1 ~ "Strongly disagree",
+#'       score == 2 ~ "Disagree",
+#'       score == 3 ~ "Neutral",
+#'       score == 4 ~ "Agree",
+#'       score == 5 ~ "Strongly agree"
+#'     )
+#'   )
+#'
+#' # While this works, it can be written more efficiently using
+#' # `recode_values()`
+#' data |>
+#'   mutate(
+#'     score = score |>
+#'       recode_values(
+#'         1 ~ "Strongly disagree",
+#'         2 ~ "Disagree",
+#'         3 ~ "Neutral",
+#'         4 ~ "Agree",
+#'         5 ~ "Strongly agree"
+#'       )
+#'   )
+#'
+#' # `recode_values()` actually has two mutually exclusive APIs. The formula API
+#' # used above, which is like `case_when()`, and a lookup style API that uses
+#' # `from` and `to` arguments. The lookup API is even better suited for this
+#' # problem, because we can move the mapping outside of the `mutate()` call
+#' # into a standalone lookup table. You could even imagine reading this
+#' # `likert` lookup table in from a separate CSV file.
+#' likert <- tribble(
+#'   ~from, ~to,
+#'   1, "Strongly disagree",
+#'   2, "Disagree",
+#'   3, "Neutral",
+#'   4, "Agree",
+#'   5, "Strongly agree"
+#' )
+#'
+#' data |>
+#'   mutate(
+#'     score = score |>
+#'       recode_values(
+#'         from = pull(likert, from),
+#'         to = pull(likert, to)
+#'       )
```

Comment:
```suggestion
#'       recode_values(from = likert$from, to = likert$to)
```

I think this is more succinct and clearer.